homeassistant/components/smart_meter_texas/config_flow.py

```diff
@@ -0,0 +1,89 @@
+"""Config flow for Smart Meter Texas integration."""
+import asyncio
+import logging
+
+from aiohttp import ClientError
+from smart_meter_texas import Account, Client
+from smart_meter_texas.exceptions import (
+    SmartMeterTexasAPIError,
+    SmartMeterTexasAuthError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    client_session = aiohttp_client.async_get_clientsession(hass)
+    account = Account(data["username"], data["password"])
+    client = Client(client_session, account)
+
+    try:
+        await client.authenticate()
+    except (asyncio.TimeoutError, ClientError, SmartMeterTexasAPIError):
+        raise CannotConnect
+    except SmartMeterTexasAuthError as error:
+        raise InvalidAuth(error)
+
+    # Return info that you want to store in the config entry.
+    return {"title": account.username}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Smart Meter Texas."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def _account_already_configured(self, account):
+        existing_accounts = {
+            entry.data[CONF_USERNAME]
+            for entry in self._async_current_entries()
+            if CONF_USERNAME in entry.data
+        }
+        return account in existing_accounts
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
+                if self._account_already_configured(user_input[CONF_USERNAME]):
+                    return self.async_abort(reason="already_configured")
+                return self.async_create_entry(title=info["title"], data=user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
```

Comment:
```suggestion
                errors["base"] = "unknown"

                if not errors:
                    if self._account_already_configured(user_input[CONF_USERNAME]):
                       return self.async_abort(reason="already_configured")
                   return self.async_create_entry(title=info["title"], data=user_input)
```

Reply:
Actually we aren't using a unique id here and calling `_abort_if_unique_id_configured` so we aren't worried about `raise data_entry_flow.AbortFlow` being trapped.

I think this is fine